### PR TITLE
Weather system. [MDB IGNORE]

### PR DESCRIPTION
**Pull request by MistakeNot4892**
## Description of changes
- Standardizes some vis_contents defines and moves vis_contents out of icon updates.
- Adds an area/turf var to track 'outside' status. Can be yes, no or area-derived. Turfs under open turfs will traverse the stack upwards until they find a roof.
- Adds a weather effect system. Weather systems are associated with a set of z-levels and will use vis_contents and a turf var (`weather`) to apply effects to turfs/mobs in their area. Weather systems also have a tick() proc that is currently unused.
- Modifies handle_environment() to always run and not always expect an airmix.

## Why and what will this PR improve
Allows us to make exterior environments more interesting and is a solid bit of work towards turfs being able to opt in and out of exterior effects/atmos.

## Authorship
Weather sprites are stolen from Polaris - not sure who originally added them.
Rest of the code is from me. Some concepts are from Polaris but none of the code is reused.
Umbrella sprites are by @PlayerDeer.

Sounds are from various authors:
Heavy rain: https://freesound.org/people/DWOBoyle/sounds/136971/
Gentle rain: https://freesound.org/people/svampen/sounds/334148/
Rain interior: https://freesound.org/people/InspectorJ/sounds/346642/
Snow: https://freesound.org/people/jgrzinich/sounds/555391/
Hail interior: https://freesound.org/people/InspectorJ/sounds/402899/
Thunder: https://freesound.org/people/straget/sounds/527664/

## Changelog

:cl:
experiment: There is now a weather system on exoplanets. Take your umbrella.
/:cl:

### CODE REVIEW

**Comment by MistakeNot4892** [comment]
Space is behaving very weirdly, possibly due to the skybox? I can't explain the visual glitching. Exoplanets seem fine though.
TqVpCJnQeU
.

**Comment by MistakeNot4892** [comment]
In the future I'd like to do some more exoplanet stuff, and also have rain add the rain material to the mob touching reagents holder, but I'm going to do that in a future PR. This system is primarily cosmetic currently.

**Comment by MistakeNot4892** [comment]
https://user-images.githubusercontent.com/2468979/138439856-51cbf1de-a86d-4dcc-9b7c-d2cd7058e96a.mp4

File: code/modules/mob/mob.dm [hunk 1]
```diff
@@ -1052,3 +1052,41 @@
...
+			LAZYADD(., brolly)
+	if(!LAZYLEN(.))
+		for(var/turf/T AS_ANYTHING in RANGE_TURFS(loc, 1))
+			for(var/obj/structure/flora/tree/tree in T)
+				if(tree.protects_against_weather)
+					LAZYADD(., tree)
+
+/mob/living/carbon/human/get_weather_protection()
+	. = ..()
+	var/obj/item/clothing/head/check_head = wear_suit
```
Comment (comma) [inline code/modules/mob/mob.dm]: `clothing/head/check_head = wear_suit`

🤔 
so head or suit?

Comment (MistakeNot4892) [reply code/modules/mob/mob.dm]: Was meant to be both but I guess I got distracted.

**Review by comma — DISMISSED** [review]

**Review by comma — APPROVED** [review]

**Pull request merged**